plugins/modules/nmcli.py

```diff
@@ -2034,6 +2040,10 @@ def connection_options(self, detect_change=False):
             options.update({
                 'infiniband.transport-mode': self.transport_mode,
             })
+            if self.infiniband_mac:
+                options.update({
+                    'infiniband.mac-address': self.infiniband_mac,
+                })
```

Comment:
I noticed that's the way it was done in the lines above and probably elsewhere in the code, but this statement is overengineered. It would be much simpler as:
```suggestion
                options['infiniband.mac-address'] = self.infiniband_mac
```